be/src/olap/parallel_scanner_builder.h

```diff
@@ -77,11 +82,19 @@ class ParallelScannerBuilder {
 
     size_t _rows_per_scanner {_min_rows_per_scanner};
 
+    /// used if `_split_by_segment_size` is true.
+    size_t _bytes_per_scanner {0};
+    size_t _min_bytes_per_scanner {32 * 1024 * 1024};
+    size_t _segment_count_factor {10};
+
+    std::map<RowsetId, std::vector<size_t>> _segments_size;
+
     std::map<RowsetId, SegmentCacheHandle> _segment_cache_handles;
 
     std::shared_ptr<RuntimeProfile> _scanner_profile;
     RuntimeState* _state;
     int64_t _limit_per_scanner;
+    bool _split_by_segment_size {false};
     bool _is_dup_mow_key;
```

Comment:
and  also init other properties such as _is_dup_mow_key _limit_per_scanner..... should init all pointer to nullptr